lib/doorkeeper/orm/active_record/application.rb

```diff
@@ -11,14 +11,9 @@ class Application < ActiveRecord::Base
     end
     has_many :authorized_applications, through: :authorized_tokens, source: :application
 
-    def self.column_names_with_table
-      self.column_names.map { |c| "#{table_name}.#{c}" }
-    end
-
     def self.authorized_for(resource_owner)
-      joins(:authorized_applications).
-        where(AccessToken.table_name => { resource_owner_id: resource_owner.id, revoked_at: nil }).
-        group(column_names_with_table.join(','))
+      where(id: Doorkeeper::AccessToken.
+        where(resource_owner_id: resource_owner.id, revoked_at: nil))
     end
```

Comment:
Sounds good. I would extract to another method or local variable the argument:

```ruby
Doorkeeper::AccessToken.where(resource_owner_id: resource_owner.id, revoked_at: nil)
```

Question: how does the scope know to grab the application id's from the `AccessToken.where` call?